### ISSUE TITLE
[Android] Stage color change causing superimposed black screen

### ISSUE THREAD

**Issue by Shaunmax** [airsdk/Adobe-Runtime-Support#1121]
On Samsung Galaxy M51,  when we changes the stage color - `Starling.current.stage.color = Color.BLUE;` its causing a black superimposed screen : -

![image](https://user-images.githubusercontent.com/16576562/131623046-241b8831-a017-4257-bf66-6b00485e015c.png)

Expected original screen : -

![image](https://user-images.githubusercontent.com/16576562/131623241-76b0533c-5e40-4268-a83a-a53c41d95d15.png)

After debugging the issue, we found that this is happening only when we use  `<containsVideo>true</containsVideo>` in the app-description xml. We have tested this issue on all other devices and no issue was reported.

To tackle this issue we have decided, not to change the stage color during the runtime.

**Comment by ajwfrost** [comment]
Thanks - the `containsVideo` thing is an interesting clue.. this looks like it may be related to some of the other black-screen issues we're seeing (as mentioned in the other thread!) which I believe we're closing in on.

Do you have an M51 on which to try this? or one of the other devices that might be affected? I'm just wondering whether we can try creating a test case and whether that then fails on one of the other Samsung devices that we have here, where we're also getting the black-screen problem...

thanks

**Comment by Shaunmax** [comment]
Yes! I have the device with me & sure, I do the testing if required.

**Comment by ajwfrost** [comment]
@Shaunmax are you able to try using the new/beta SDK that we've just uploaded, to see if this helps?
https://airsdk.harman.com/download/33.1.1.620

thanks

**Comment by Shaunmax** [comment]
> @Shaunmax are you able to try using the new/beta SDK that we've just uploaded, to see if this helps?
> https://airsdk.harman.com/download/33.1.1.620
> 
> thanks

@ajwfrost  I tested it & the issue persists...

**Comment by ajwfrost** [comment]
Okay thanks for checking - it's definitely something to do with the layering of surfaces though! Curious/frustrating that it only happens on some devices...

**Comment by ajwfrost** [comment]
@Shaunmax is it possible to get a small test case that demonstrates this problem please, built with "apk-debug" setting? And it would be good to know if the elements that are showing up with the blue border around are Starling ones or are just normal display list objects?

thanks

**Comment by Shaunmax** [comment]
@ajwfrost i am not sure how to demonstrate it ? 

> built with "apk-debug" setting?

yes, for testing I used a debug build 

> And it would be good to know if the elements that are showing up with the blue border around are Starling

yes all Starling objects 

I am sharing the code : - 

`package {
import feathers.controls.Button;
import feathers.controls.ScrollBarDisplayMode;
import feathers.controls.ScrollPolicy;
import feathers.controls.ScrollText;
import feathers.controls.TextInput;

import starling.core.Starling;
import starling.display.Quad;

import starling.display.Sprite;
import starling.events.Event;
import starling.text.TextFormat;
import starling.utils.Align;
import starling.utils.Color;

public class TestBlackScreen extends Sprite{

    private var num1Text:TextInput;
    private var num2Text:TextInput;
    private var chatText:TextInput;
    private var resultText:TextInput;
    private var joinBtn:Button;
    private var addBtn:Button;
    private var senBtn:Button;
    private var buyBtn:Button;
    public var chatTextField:ScrollText;

    public function TestBlackScreen() {
        joinBtn = new Button();
        //joinBtn.width = 80;
        //joinBtn.height = 30;
        joinBtn.fontStyles = new TextFormat("Helvetica", 16, Color.BLACK);
        joinBtn.label = "JOIN";
        joinBtn.addEventListener(Event.TRIGGERED, join_click);
        addChild(joinBtn);
        joinBtn.validate();
        joinBtn.x = Starling.current.stage.stageWidth/2 - (joinBtn.width);
        joinBtn.y = 10;

        num1Text = new TextInput();
        num1Text.fontStyles = new TextFormat("Helvetica", 16, Color.WHITE);
        num1Text.width = 90;
        num1Text.height = 30;
        num1Text.x = 10;
        num1Text.y = joinBtn.y + joinBtn.height + 20;
        num1Text.prompt = "Enter N1";
        addChild(num1Text);
        num1Text.validate();

        num2Text = new TextInput();
        num2Text.fontStyles = new TextFormat("Helvetica", 16, Color.WHITE);
        num2Text.width = 90;
        num2Text.height = 30;
        num2Text.x = num1Text.x + num1Text.width + 10;
        num2Text.y = num1Text.y;
        num2Text.prompt = "Enter N2";
        addChild(num2Text);
        num2Text.validate();

        addBtn = new Button();
        addBtn.x = num2Text.x + num2Text.width + 10;
        addBtn.y = num2Text.y;
        //addBtn.width = 80;
        //addBtn.height = 30;
        addBtn.fontStyles = new TextFormat("Helvetica", 16, Color.BLACK);
        addBtn.label = "ADD";
        addBtn.addEventListener(Event.TRIGGERED, add_click);
        addChild(addBtn);

        resultText = new TextInput();
        resultText.fontStyles = new TextFormat("Helvetica", 18, Color.WHITE);
        resultText.width = 90;
        resultText.height = 30;
        resultText.x = 10;
        resultText.y = num1Text.y + num1Text.height + 10;
        resultText.prompt = "Result";
        addChild(resultText);
        resultText.validate();

        buyBtn = new Button();
        buyBtn.x = resultText.x + resultText.width + 10;
        buyBtn.y = resultText.y;
        //addBtn.width = 80;
        //addBtn.height = 30;
        buyBtn.fontStyles = new TextFormat("Helvetica", 16, Color.BLACK);
        buyBtn.label = "BUY";
        buyBtn.addEventListener(Event.TRIGGERED, buy_click);
        addChild(buyBtn);

        chatText = new TextInput();
        chatText.fontStyles = new TextFormat("Helvetica", 18, Color.WHITE);
        chatText.width = 190;
        chatText.height = 30;
        chatText.x = 10;
        chatText.y = resultText.y + resultText.height + 10;
        chatText.prompt = "Say something..";
        addChild(chatText);
        chatText.validate();

        senBtn = new Button();
        senBtn.x = chatText.x + chatText.width + 10;
        senBtn.y = chatText.y;
        //senBtn.width = 80;
        //senBtn.height = 30;
        senBtn.fontStyles = new TextFormat("Helvetica", 16, Color.BLACK);
        senBtn.label = "SEND";
        senBtn.addEventListener(Event.TRIGGERED, send_click);
        addChild(senBtn);

        var quad:Quad = new Quad(Starling.current.stage.stageWidth - 20, Starling.current.stage.stageHeight - (chatText.y + chatText.height + 40), Color.MAROON);
        addChild(quad);
        quad.x = 10;
        quad.y = chatText.y + chatText.height + 10;

        chatTextField = new ScrollText();
        chatTextField.fontStyles = new TextFormat("Helvetica", 12, Color.WHITE, Align.LEFT);
        //chatTextField.backgroundColor = Color.GRAY;
        //chatTextField.border = true;
        //chatTextField.background = true;
        chatTextField.horizontalScrollPolicy = ScrollPolicy.OFF;
        chatTextField.hasElasticEdges = true;
        chatTextField.width = Starling.current.stage.stageWidth - 10;
        chatTextField.height = Starling.current.stage.stageHeight - (chatText.y + chatText.height + 40);
        chatTextField.scrollBarDisplayMode = ScrollBarDisplayMode.FIXED;
        this.addChild(chatTextField);
        chatTextField.x = 10;
        chatTextField.y = chatText.y + chatText.height + 10;
    }

    private function add_click(event:Event):void {

    }

    private function buy_click(event:Event):void {

    }

    private function send_click(event:Event):void {

    }

    private function join_click(event:Event):void {

    }
}
}
`

**Comment by ajwfrost** [comment]
All Samsung devices, or just some? We have a few different ones here including ones that are meant to exhibit some of these odd black screen related problems..
If you can upload the APK to the below link it would be great, then we can check that we reproduce it!
https://transfer.harman.com/requests/LxdVtmRFcqSVcjd4iGhC34
thanks

**Comment by Shaunmax** [comment]
> All Samsung devices, or just some? We have a few different ones here including ones that are meant to exhibit some of these odd black screen related problems.. If you can upload the APK to the below link it would be great, then we can check that we reproduce it! https://transfer.harman.com/requests/LxdVtmRFcqSVcjd4iGhC34 thanks

I haven't tested this on many Samsung devices, for the time being I could only reproduce it on just two devices!
